velox/exec/Unnest.cpp

```diff
@@ -111,56 +112,111 @@ RowVectorPtr Unnest::getOutput() {
   }
 
   const auto size = input_->size();
-  const auto maxOutputSize = outputBatchRows();
 
   // Limit the number of input rows to keep output batch size within
-  // 'maxOutputSize' if possible. Process each input row fully. Do not break
-  // single row's output into multiple batches.
-  vector_size_t numInput = 0;
+  // 'maxOutputSize' if possible. Not process each input row fully when a
+  // row's output exceeds 'maxOutputSize'. The output of single row may be
+  // divided into multiple batches.
   vector_size_t numElements = 0;
-  for (auto row = nextInputRow_; row < size; ++row) {
-    numElements += rawMaxSizes_[row];
-    ++numInput;
-
-    if (numElements >= maxOutputSize) {
-      break;
-    }
-  }
-
+  vector_size_t partialProcessRowStart = -1;
+  auto rowRange = extractRowRange(size, numElements, partialProcessRowStart);
```

Comment:
Would you document the behavior for 'extractRowRange' on when the passed `partialProcessRowStart` remains unchanged?